pyccel/ast/numpy_wrapper.py

```diff
@@ -0,0 +1,283 @@
+#------------------------------------------------------------------------------------------#
+# This file is part of Pyccel which is released under MIT License. See the LICENSE file or #
+# go to https://github.com/pyccel/pyccel/blob/master/LICENSE for full license details.     #
+#------------------------------------------------------------------------------------------#
+
+"""
+Handling the transitions between python code and C code using (Numpy/C Api).
+"""
+
+import numpy as np
+
+from .datatypes         import (NativeInteger, NativeReal, NativeComplex,
+                                NativeBool, NativeGeneric, NativeVoid)
+
+from .cwrapper          import PyccelPyObject, PyccelPyArrayObject
+
+from .core              import FunctionDef, FunctionCall
+from .operators         import PyccelNot, PyccelEq
+
+
+from .literals          import LiteralInteger
+
+from .variable          import Variable
+
+from ..errors.errors   import Errors
+from ..errors.messages import PYCCEL_RESTRICTION_TODO
+
+errors = Errors()
+
+__all__ = (
+    #------- CAST FUNCTIONS ------
+    'pyarray_to_c_ndarray',
+    #-------CHECK FUNCTIONS ------
+    'array_checker',
+    #-------HELPERS ------
+    'numpy_get_dim',
+    'numpy_get_ndims',
+    'numpy_get_data',
+    #-------OTHERS--------
+    'get_numpy_max_acceptable_version_file'
+)
+
+
+def get_numpy_max_acceptable_version_file():
+    """
+    Get the macro specifying the last acceptable numpy version. If numpy is more
+    recent than this then deprecation warnings are shown.
+
+    The last acceptable numpy version is 1.19. If the current version is older
+    than this then the last acceptable numpy version is the current version
+    """
+    numpy_max_acceptable_version = [1, 19]
+    numpy_current_version = [int(v) for v in np.version.version.split('.')[:2]]
+    numpy_api_macro = '# define NPY_NO_DEPRECATED_API NPY_{}_{}_API_VERSION\n'.format(
+        min(numpy_max_acceptable_version[0], numpy_current_version[0]),
+	    min(numpy_max_acceptable_version[1], numpy_current_version[1]))
```

Comment:
I don't think this is a good way to handle Major and Minor versions, those items should not be separated.
another thing is that I understand from numpy C API is that the `NPY_NO_DEPRECATED_API` mechanism should be used with the new version realized, but I see here the opposite.

Reply:
You are right about not splitting the major and minor versions. Fixed in 754fd1d

Reply:
The flag should be used with the newest version that is tested (numpy 1.19 for us), however, if you are running numpy 1.18 then the macro `NPY_1_19_API_VERSION` is not defined so an error will be raised. Instead we need to use `NPY_1_18_API_VERSION` so the macro exists. I believe there are no changes to the numpy wrapper since numpy 1.7 so it is not a problem if the macro defines a lower version.

Numpy version installed | macro used
----|----
1.10 | `NPY_1_10_API_VERSION`
1.18 | `NPY_1_18_API_VERSION`
1.19 | `NPY_1_19_API_VERSION`
1.20 | `NPY_1_19_API_VERSION`
1.24 | `NPY_1_19_API_VERSION`

Reply:
what is the case for the opposite, if the user uses a higher version than 1.19?